fpmsyncd/routesync.cpp

```diff
@@ -747,6 +754,79 @@ void RouteSync::onRouteMsg(int nlmsg_type, struct nl_object *obj, char *vrf)
 }
 
 /* 
+ * Handle label route
+ * @arg nlmsg_type      Netlink message type
+ * @arg obj             Netlink object
+ */
+void RouteSync::onLabelRouteMsg(int nlmsg_type, struct nl_object *obj)
+{
+    struct rtnl_route *route_obj = (struct rtnl_route *)obj;
+    struct nl_addr *daddr;
+    char destaddr[MAX_ADDR_SIZE + 1] = {0};
+
+    daddr = rtnl_route_get_dst(route_obj);
+    nl_addr2str(daddr, destaddr, MAX_ADDR_SIZE);
+    SWSS_LOG_INFO("Receive new route message dest addr: %s", destaddr);
+    if (nl_addr_iszero(daddr)) return;
+
+    if (nlmsg_type == RTM_DELROUTE)
+    {
+        m_label_routeTable.del(destaddr);
+        return;
+    }
+    else if (nlmsg_type != RTM_NEWROUTE)
+    {
+        SWSS_LOG_INFO("Unknown message-type: %d for %s", nlmsg_type, destaddr);
+        return;
+    }
+
+    switch (rtnl_route_get_type(route_obj))
+    {
+        case RTN_BLACKHOLE:
+        {
+            vector<FieldValueTuple> fvVector;
+            FieldValueTuple fv("blackhole", "true");
+            fvVector.push_back(fv);
+            m_label_routeTable.set(destaddr, fvVector);
+            return;
+        }
+        case RTN_UNICAST:
+            break;
+
+        case RTN_MULTICAST:
+        case RTN_BROADCAST:
+        case RTN_LOCAL:
+            SWSS_LOG_INFO("BUM routes aren't supported yet (%s)", destaddr);
+            return;
+
+        default:
+            return;
+    }
+
+    struct nl_list_head *nhs = rtnl_route_get_nexthops(route_obj);
+    if (!nhs)
+    {
+        SWSS_LOG_INFO("Nexthop list is empty for %s", destaddr);
+        return;
+    }
+
+    /* Get nexthop lists */
+    string nexthops = getNextHopList(route_obj);
+    string ifnames = getNextHopIf(route_obj);
```

Comment:
It's optional, but I think these two should be in one API that returns a list of <Nexthop> Object, which can have "getNexthopList().str()" and "getNextHopIf().str()".